qt/scientific_interfaces/test/ISISReflectometry/MainWindow/MainWindowPresenterTest.h

```diff
@@ -162,6 +163,36 @@ class MainWindowPresenterTest : public CxxTest::TestSuite {
     verifyAndClear();
   }
 
+  void testWarningGivenIfRemoveUnsavedBatch() {
+    auto optionsDialogPresenter =
+        std::make_unique<NiceMock<MockOptionsDialogPresenter>>();
+    auto optionsDialogPresenterRaw = optionsDialogPresenter.get();
+    auto presenter = makePresenter(std::move(optionsDialogPresenter));
+    auto const batchIndex = 0;
+    expectBatchIsNotAutoreducing(batchIndex);
+    expectBatchIsNotProcessing(batchIndex);
+    expectWarnDiscardChanges(optionsDialogPresenterRaw, true);
```

Comment:
Again I don't think there are tests for the inverse case where the warning is disabled, i.e. check for false here?